### PR TITLE
End game on skip and forceskip

### PR DESCRIPTION
**Pull request by taahamahdi**
Check for gameFinished inside each endRound call

- Replaces the check inside setTimeout
- No longer need the check inside ,forceskip and ,skip

### CODE REVIEW

File: src/models/game_session.ts [hunk 1]
```diff
@@ -104,6 +104,13 @@ export default class GameSession {
             this.connection.removeAllListeners();
         }
         this.stopGuessTimeout();
+        if (this.gameType === GameType.ELIMINATION) {
+            const eliminationScoreboard = this.scoreboard as EliminationScoreboard;
```
Comment (Brainicism) [inline src/models/game_session.ts:108]: Do we need this cast?

Comment (Brainicism) [reply src/models/game_session.ts:108]: We can make the `gameFinished` check for classic mode in here as well

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/models/game_session.ts (hunk 1 above)
Comment (taahamahdi) [reply src/models/game_session.ts:108]: The cast was needed to call `eliminationScoreboard`'s `gameFinished` method instead of `Scoreboard`'s

Comment (taahamahdi) [reply src/models/game_session.ts:108]: > We can make the `gameFinished` check for classic mode in here as well

To do this, we need to access `guildPreference` for `isGoalSet`, so we need to call `getGuildPreference`. However, to await this, I need to turn endRound into an asynchronous method. Is that okay?

**Review by Brainicism — APPROVED** [review]

**Pull request merged**